backend/src/middlewares/auth/strategies/google.strategy.ts

```diff
@@ -0,0 +1,75 @@
+import { ENV } from '@common/enums/app/env.enum';
+import { prisma } from '@data/prisma-client';
+import { verifyToken } from '@helpers/helpers';
+import { JwtPayload } from 'jsonwebtoken';
+import { Strategy as GoogleStrategy } from 'passport-google-oauth20';
+
+const googleStrategy = new GoogleStrategy(
+  {
+    clientID: ENV.GOOGLE.CLIENT_ID as string,
+    clientSecret: ENV.GOOGLE.CLIENT_SECRET as string,
+    callbackURL: `${ENV.API.V1_PREFIX}/auth/google/redirect`,
+    passReqToCallback: true,
+    scope: ['profile', 'email'],
+  },
+  async (req, accessToken, refreshToken, profile, done) => {
+    let user;
+    try {
+      user = await prisma.user.findFirst({
+        where: {
+          User_Security: {
+            google_acc_id: profile.id,
+          },
+        },
+      });
+      if (!user) {
```

Comment:
Is this statement correct? Why do we try to update the `user` if it wasn't found?

Reply:
Well, user wasn't found by Google ID, but if we recieved jwt it means that user exists and can be found by uuid so we can simply add Google ID to profile